labs/eventarc-apihub/lab.md

```diff
@@ -0,0 +1,194 @@
+    --event-filters="type=google.cloud.audit.log.v1.written" \
+    --event-filters="serviceName=apigee.googleapis.com" \
+    --event-filters="methodName=google.cloud.apigee.v1.DeploymentService.DeployApiProxy"
+```
+
+## Step 1: An API Developer creates/updates an API in the hub
+
+Customers have shown a great interest in automating at scale the publication of API proxies in their fully managed Apigee X organization in correspondence of the publication from the developers of the specifications docs onto the Apigee Api hub; the API proxies could be generated autonomously either as pass-through proxies (or as compliant and pre-populated by a given set of pre-configured Policy configurations and even a standard template including flow callouts to common organization and/or environment level resources).
+
+We want to define a sequence of events that can be autonomously generated and handled in occurrence of each. These events correspond to the first entry point for the API hub emitting an Eventarc trigger when an API developer interacts with the Api hub user interface; we will describe what happens next in Step 2.
+
+For simplicity, the high level detail above shows the scenario where the API Proxy bundle being a straightforward pass-through proxy, i.e. the proxy bundle does not exist yet and does not need to be retrieved in advance of its manipulation (it is simply created without policies) from the GIT Repository.
+
+## Step 2: Api hub emits an Eventarc trigger
+Api hub emits an Eventarc trigger event upon one of the configured events that we want an automation task to act upon (according to step 1); we will describe in greater detail in the following steps what actions are initiated according to such conditions being met. Each individual trigger event will initiate a new instance of the Workflows execution.
+
+## Step 3: An instance of GCP Workflows is started
+All orchestration presented here is easily implemented using configuration in Google Cloud Workflows. The workflow engine is serverless and runs in Google Cloud; it is hence a cloud native tool that can be configured to initiate workflows on receiving an Eventarc specific trigger. 
+
+Workflows is configured to initiate when the Eventarc trigger coming from API hub on new events corresponding  to either:
+- the creation of a new API in Api hub starting from its SOAP WSDL or OpenAPI specs;
+- the update of an existing API in Api hub, when a new version for its OpenAPI specs have been uploaded or - for example - when the corresponding SOAP WSDL file has been amended with a new version (or, for REST API endpoints, a different OpenAPI specification file is uploaded).
+
+Also note that an alternative (separate) entry point for the orchestration pipelines described in this article is the publication from an authorized Apigee API management console (at https://apigee.google.com) user of either a new API proxy or a new version of an existing API proxy.
```

Comment:
Please remove the link in this line, as I can see the GCP Console eventually fully replacing the standalone Apigee-only console/UI.